### PR TITLE
Oz - fix recursion/path params bugs

### PR DESCRIPTION
**Pull request by ozhey**

### CODE REVIEW

File: plugin/plugin.go [hunk 1]
```diff
@@ -452,6 +454,19 @@ func getParamDefault(defaultValue interface{}, paramType string) string {
 	return paramDefault
 }
 
+func hasDuplicates(path string) bool {
+	paramsArray := strings.Split(path,".")
```
Comment (guyblink) [inline plugin/plugin.go:458]: Use a the const consts.BodyParamDelimiter

File: plugin/plugin.go [hunk 2]
```diff
@@ -362,7 +362,9 @@ func handleBodyParams(schema *openapi3.Schema, parentPath string, action *plugin
 		if parentPath != "" {
 			fullParamPath = parentPath + consts.BodyParamDelimiter + fullParamPath
 		}
-
+		if hasDuplicates(fullParamPath) {
+			 continue
```
Comment (guyblink) [inline plugin/plugin.go:366]: Did this work?
I mean, the way I see it we should stop at duplicate, add the value at this point and then continue to the next param.
Just `continue` will never add the value, no?

I guess what we need here is:
`			actionParam := parseActionParam(action.Name, fullParamPath, bodyProperty, isParamRequired)
			action.Parameters[fullParamPath] = actionParam`

and then:
`continue`.

File: plugin/handlers/parsers.go [hunk 3]
```diff
@@ -35,7 +35,20 @@ func DefineOperations(openApi *openapi3.T) error {
 
 			// All the parameters required by a handler are the union of the
 			// global parameters and the local parameters.
-			allParams := append(globalParams, localParams...)
+			allParams := append(localParams, globalParams...)
+
+			// Remove any duplicate names in the params array. The last occurrence will be removed
+			// because we want to prioritize the op param and not the global param
+			paramExists := make(map[string]bool)
```
Comment (guyblink) [inline plugin/handlers/parsers.go:42]: Not sure I understand. What do you do with paramExists eventually?

File: plugin/handlers/parsers.go [hunk 4]
```diff
@@ -35,7 +35,20 @@ func DefineOperations(openApi *openapi3.T) error {
...
+			i := 0 // output index
+			for _, x := range allParams {
+				if !paramExists[x.ParamName] {
+					allParams[i] = x
+					i++
+					paramExists[x.ParamName] = true
+				}
+			}
+			allParams = allParams[:i]
```
Comment (guyblink) [inline plugin/handlers/parsers.go:51]: Maybe only add globalParams that don't exist in localParams to allParams?

**Review by guyblink — CHANGES_REQUESTED** [review]

File: plugin/handlers/parsers.go (hunk 4 above)
Comment (ozhey) [reply plugin/handlers/parsers.go:51]: It's exactly what it does - the localParams come first in the array so they are added first.
If later arrives a parameter with the same name (from the globalParams or the localParams, it doesn't matter), it won't be added thanks to the conditional:        if !paramExists[x.ParamName]

File: plugin/handlers/parsers.go (hunk 3 above)
Comment (ozhey) [reply plugin/handlers/parsers.go:42]: Nothing, it just helps me keeping track of which parameter names are already in use.
The loop that comes right after this line basically removes any parameter with a name thats already being used.

File: plugin/plugin.go (hunk 2 above)
Comment (ozhey) [reply plugin/plugin.go:366]: The hasDuplicates() function only returns true when there's already a duplicate, so I think we don't want to add the value.
For example, if the full path is currently one.two.three.two - we don't want to add the 2nd two, right?
What do you think?

File: plugin/plugin.go (hunk 1 above)
Comment (ozhey) [reply plugin/plugin.go:458]: Will do

File: plugin/handlers/parsers.go [hunk 5]
```diff
@@ -35,7 +35,19 @@ func DefineOperations(openApi *openapi3.T) error {
 
 			// All the parameters required by a handler are the union of the
 			// global parameters and the local parameters.
-			allParams := append(globalParams, localParams...)
+			allParams := []parameterDefinition{}
+			for _, globalParam := range globalParams {
+				exists := false
+				for _, localParam := range localParams {
+					if globalParam.ParamName == localParam.ParamName {
+						exists = true
```
Comment (guyblink) [inline plugin/handlers/parsers.go:43]: you can add `break` after this

**Review by guyblink — CHANGES_REQUESTED** [review]

File: plugin/handlers/parsers.go (hunk 5 above)
Comment (ozhey) [reply plugin/handlers/parsers.go:43]: Done

**Pull request merged**